### PR TITLE
Build warnings

### PR DESCRIPTION
**Pull request by remicollet**
Too much build warnings introduce noise which can hide serious issues.
This fix all of them, even if some are mostly false-positive from the compiler.

### CODE REVIEW

**Comment by remicollet** [comment]
Of course, need to be reviewed carefully.
and yes, I build everything using -Wall ;)

File: php_mustache.cpp [hunk 1]
```diff
@@ -79,7 +79,7 @@ zend_module_entry mustache_module_entry = {
   (char *) PHP_MUSTACHE_NAME,    /* Name */
   NULL,                          /* Functions */
   PHP_MINIT(mustache),           /* MINIT */
-  NULL,                          /* MSHUTDOWN */
+  PHP_MSHUTDOWN(mustache),       /* MSHUTDOWN */
```
Comment (jbboehr) [inline php_mustache.cpp]: Well, that was bad.

**Comment by jbboehr** [comment]
LGTM

**Comment by remicollet** [comment]
Thanks

**Pull request merged**